test/MetaStable.integration.spec.ts

```diff
@@ -0,0 +1,339 @@
+// yarn test:only test/MetaStable.integration.spec.ts
+import dotenv from 'dotenv';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { defaultAbiCoder } from '@ethersproject/abi';
+import { BalancerHelpers__factory } from '@balancer-labs/typechain';
+import { vaultAddr } from './testScripts/constants';
+import { SubgraphPoolBase, SOR } from '../src';
+import {
+    Network,
+    MULTIADDR,
+    SOR_CONFIG,
+    ADDRESSES,
+} from './testScripts/constants';
+import { OnChainPoolDataService } from './lib/onchainData';
+import { TokenPriceService } from '../src';
+import { AddressZero } from '@ethersproject/constants';
+import { BigNumber, formatFixed, parseFixed } from '@ethersproject/bignumber';
+import { expect } from 'chai';
+import { MetaStablePool } from '../src/pools/metaStablePool/metaStablePool';
+
+dotenv.config();
+
+// Balancer USDC-USDT Oracle Stable Pool
+const testPool: SubgraphPoolBase = {
+    id: '0x9f383f91c89cbd649c700c2bf69c2a828af299aa0002000000000000000003a6',
+    address: '0x9f383f91c89cbd649c700c2bf69c2a828af299aa',
+    swapEnabled: true,
+    poolType: 'MetaStable',
+    swapFee: '0.000001',
+    totalWeight: '0',
+    totalShares: '1.501763384758054657',
+    tokensList: [
+        '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+        '0xdac17f958d2ee523a2206206994597c13d831ec7',
+    ],
+    tokens: [
+        {
+            address: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+            balance: '1.129827',
+            decimals: 6,
+            priceRate: '0.9992',
+            weight: null,
+        },
+        {
+            address: '0xdac17f958d2ee523a2206206994597c13d831ec7',
+            balance: '0.382611',
+            decimals: 6,
+            priceRate: '1',
+            weight: null,
+        },
+    ],
+};
+
+let sor: SOR;
+const networkId = Network.MAINNET;
+const { ALCHEMY_URL: jsonRpcUrl } = process.env;
+const rpcUrl = 'http://127.0.0.1:8545';
+const provider = new JsonRpcProvider(rpcUrl, networkId);
+let subgraphPoolDataService: OnChainPoolDataService;
+
+// Setup SOR with data services
+function setUp(networkId: Network, provider: JsonRpcProvider): SOR {
```

Comment:
This is pretty much the same as used in ComposableStable. Can we create a common helper.